src/workerd/api/trace.h

```diff
@@ -74,7 +77,10 @@ class TraceItem final: public jsg::Object {
                     jsg::Ref<QueueEventInfo>,
                     jsg::Ref<EmailEventInfo>,
                     jsg::Ref<TailEventInfo>,
-                    jsg::Ref<CustomEventInfo>> EventInfo;
+                    jsg::Ref<CustomEventInfo>,
+                    jsg::Ref<HibernatableWebSocketMessageEventInfo>,
```

Comment:
Do we really need 3 different objects here?
Can't we do like `FetchEventInfo` or `TailEventInfo` and have something like HibernatableWebSocketEventInfo::Message / HibernatableWebSocketEventInfo::Error / HibernatableWebSocketEventInfo::Close ?

Reply:
https://github.com/cloudflare/workerd/pull/709/commits/cee21cb1f87d722f9423fa145ea6051a967f6b44 I added it as a separate commit since I don't want to deal with git if we want to drop it.

I suspect the reason was because those other events have several properties you might want to access, so it's nice to store them in the same place. Our `HibernatableWebSocketEventInfo` really just has 1 property that can have several different states (Message/Close/Error). Having them all together makes it slightly more cumbersome in JS because you have to go through an additional layer of indirection to get the event type you care about. Ex. `event.webSocketEventType` now becomes `event.getWebSocketEvent.webSocketEventType`.